components/camel-zipfile/src/main/java/org/apache/camel/dataformat/zipfile/ZipFileDataFormat.java

```diff
@@ -48,16 +48,17 @@ public String getDataFormatName() {
 
     @Override
     public void marshal(final Exchange exchange, final Object graph, final OutputStream stream) throws Exception {
-        String filename = exchange.getIn().getHeader(FILE_NAME, String.class);
-        if (filename == null) {
+        String filename;
+        String filepath = exchange.getIn().getHeader(FILE_NAME, String.class);
+        if (filepath == null) {
             // generate the file name as the camel file component would do
-            filename = StringHelper.sanitize(exchange.getIn().getMessageId());
+            filename = filepath = StringHelper.sanitize(exchange.getIn().getMessageId());
```

Comment:
Double assigment? Intentional?

Reply:
Yes, this is intentional when there is no filename set, the generated filename should then be used to name the zip entry aswell.

Reply:
This line is modified that s why i asked.

Now it is like 

filename = filepath = exchange...

Reply:
Unless you explained, i did not get that. Seems reasonable. Cant think of complicated cases out of the first look. Seems ok.if somebody else also review, it would be good